server/server.go

```diff
@@ -306,10 +313,7 @@ func (s *server) invokeHandleEndpoint() endpoint.Endpoint {
 		ri := rpcinfo.GetRPCInfo(ctx)
 		methodName := ri.Invocation().MethodName()
 		serviceName := ri.Invocation().ServiceName()
-		svc := s.svcs.getService(serviceName)
-		if svc == nil {
-			svc = s.svcs.defaultSvc
-		}
+		svc := s.svcs.svcMap[serviceName]
```

Comment:
Please check again if svc will be ensured to be non-nil by some logic before; if not, please add a check and return a specific error (instead of panic later with NPE)